frameworks/TunedRandomForest/exec.py

```diff
@@ -33,14 +35,28 @@ def pick_values_uniform(start: int, end: int, length: int):
     return sorted(set([int(f) for f in uniform_floats]))
 
 
+def extrapolate_with_worst_case(values: List[float], n: int = 5) -> float:
+    """ Extrapolate the next value for `values`, based on the last `n` samples. """
+    n = min(len(values), n)
+    return values[-1] + max(v_next - v_prev for v_prev, v_next in zip(values[-n:], values[-n+1:]))
+
+
 def run(dataset, config):
     log.info(f"\n**** Tuned Random Forest [sklearn v{sklearn.__version__}] ****\n")
 
     is_classification = config.type == 'classification'
 
-    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
-    tuning_params = config.framework_params.get('_tuning', training_params)
+    training_params = {
+        k: v for k, v in config.framework_params.items()
+        if not (k.startswith('_') or k == "n_estimators")
+    }
+    if "max_features" in training_params:
+        raise ValueError("`max_features` may not be specified for Tuned Random Forest.")
+
     n_jobs = config.framework_params.get('_n_jobs', config.cores)  # useful to disable multicore, regardless of the dataset config
+    step_size = config.framework_params.get('_step_size', 10)
```

Comment:
I assume that the '10' here is a fallback value, in case the step size is not defined in the config?

Reply:
It is. And for now we will keep the framework definitions identical from the previous iteration, meaning that this default value will be the one used in tests. I did some small scale testing and it seemed that fitting `10` or `100` trees at once did not make a meaningful difference to the end result, so I decided to keep it small (one could argue that maybe it should be `1` instead but even `Higgs` fit hundreds of trees).